sdk/storage/azure-storage-common/inc/azure/storage/common/version.hpp

```diff
@@ -16,7 +16,7 @@
 #define AZURE_STORAGE_COMMON_VERSION_PATCH 0
 #define AZURE_STORAGE_COMMON_VERSION_PRERELEASE "beta.10"
 
-namespace Azure { namespace Storage { namespace Common {
+namespace Azure { namespace Storage { namespace Common { namespace _detail {
```

Comment:
Same as KeyVault, wondering if `StorageCommonVersion` instead of Storage::Common::_detail`  is better.
Of maybe Storage::_detail::CommonVersion

Reply:
Right now, end-customers can see this namespace as
Azure::Storage::Common::_detail
When using any Storage service

Reply:
This is the one place not having a common namespace is tough.  Ideally we want <service>::PackageVersion for each package.  Also I expect PackageVersion is never used from the common package.  We don't however want a common namespace in any service.  Common is confusing to the user.  We should make an exception for the ::Common::_detail namespace as that is details for the common package.

Reply:
The exception is fine, just notice that an end-user typing `Azure::Storage::` will see the Common namespace populated from intellisense. Just for latter find out that Azure::Storage::Common::  will lead to `_detail` only.  
That's why I would move the common version out of Common